hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReader.java

```diff
@@ -200,6 +332,71 @@ public HoodieLogBlock next() {
     return nextBlock;
   }
 
+  /**
+   * hasPrev is not idempotent
+   *
+   * @return
+   */
+  public boolean hasPrev() {
+    try {
+      reverseLogFilePosition = lastReverseLogFilePosition;
+      reverseLogFilePosition -= Long.BYTES;
+      lastReverseLogFilePosition = reverseLogFilePosition;
+      inputStream.seek(reverseLogFilePosition);
+    } catch (Exception e) {
+      // Either reached EOF while reading backwards or an exception
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * This is a reverse iterator
+   * Note: At any point, an instance of HoodieLogFormatReader should either iterate reverse (prev)
+   * or forward (next). Doing both in the same instance is not supported
+   * WARNING : Every call to prev() should be preceded with hasPrev()
```

Comment:
it is, that's what I wanted to do but the new iterator needs to hold context passed by the user etc. so kept this change local for now since this PR is more about LogFormat V2.

Reply:
okay.. general comment:  please file a follow up task from all the left over cleanups here.. It will take significant amount of time to figure these little things out again